### ISSUE TITLE
Fix false positives for spaceless operators in unit-case

### ISSUE THREAD

**Issue by superchow** [stylelint/stylelint#5083]
> Clearly describe the bug

I want my Less variable to be itself，but it always changed to lowercase.

> Which rule, if any, is the bug related to?

 `value-keyword-case`

> What code is needed to reproduce the bug?

```css
.u-input(@lineHeight, @paddingTop, @paddingBottom) {
  height: 1px*@lineHeight;
  padding-top: 1px*@paddingTop;
  padding-bottom: 1px*@paddingBottom;
  line-height: 1px*@lineHeight;
}
```

> What stylelint configuration is needed to reproduce the bug?

```json
{
  "rules": {
    "value-keyword-case":  ["lower", {
      "ignoreTypes": [/@\w+/]
    }],
  }
}
```

> Which version of stylelint are you using?

`12.0.0`

> How are you running stylelint: CLI, PostCSS plugin, Node.js API?

"CLI with `stylelint "**/*.css" --config myconfig.json`"

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

"Yes, it's related to Less nested properties."

> What did you expect to happen?

`@lineHeight` should not to be `@lineheight` In the example, after run `stylelint "**/*.css" --config myconfig.json`. How to keep it self ?

**Comment by jeddy3** [comment]
@superchow Thanks for the report and for using the template.

It looks like it's the `unit-case` rule causing the issue. See [this demo](https://stylelint.io/demo#N4Igxg9gJgpiBcIB0BXAtASwHYAcUBcAKAAQBtsYAJGDAcwAt8AaAAmJwEMoptaAVCDlbsuPLLQBCEfPggBbAJQtgAHSwsW9Gg3zwWARhwAPAFRkK1OowDcajZ2680snHsOmRj8QJy31LBzFaNAAjaVk5N2MzQN4pGXk-DXIsGDQtK10DaPNUyx0-AF81ECYQAGcATyx8DiMEEFIYcvLS8AgsADM6BtV-FRAAJxQm8oG9Po0NAY58NGGmtCwINBQsAGtlgHcscZZ8YZgmOymBkNIIMHWllZg5HHxKvYOUI5Pp9ovBm8wsADcOOQoOkYEZnodjv4PpA5HIYDUfncHk8EPsIe8WANYGBSBxBrMMB1Qhcrj8oCgcOQwLM0jhBoIYIN8BhmnsANoYjQvN5QqaTKYCgZ0ZaDGDszkCzHtLDlGBgAgYP5pcmUjDU-DNNBbDD4ehoHidTqM+FzAGkV5jEASjQAXQlxV5NshgpA2Nx+OZRPOl2uyzQ5XoECZ9A4WGBdIZTMqaAgSsGgwwsEtem5ztOIE6HTmnQ4cgwpGjfpVVJpS1zrNRqYxA0zCJzeYLPzzLSctHhjLVaHr+ej6xglS2Qag4NeaY+nTWYE9WDQ1NIYB+2DNiZHPJdE6wU8JM7nC79a3KnDAMGBkdmQdXY6lG63RJSMDxaFo+J4Jp+yxltTDeOBPFFt92St0V5AY+0qTp8ThfU5XdAkiT9DB7iDL98EvasQDhHgOC7B98BQUUyyg-cNm2QCU2Al0i2aY8wycQ85QwbowB1FFyNHdCiwpEsNTQWZMCQpl5hGCs2LXdNONVdU0llJopyDZM0XYkCQD9JFHn9CB8OPNDlNUowDmw2U80gC4ZR0yiViXQFE31TTzmk3EA1neQ4RqBSq2UiMcEZdTiM2CAdnM9MZLlWRvhwWUUCgFYcQ4FofjWfzAqApSXRCuTwsi6K0BgJpXLmPzSKCj50rC5xKm8hKSICsjFLEkqDicP1Ui2e9iqlNYdSqpLao8l0+IWNI1Oje9QhgTNRXFXkPkBLYOEqS0rw0flJSlUFjweKbVpdb0rlGcpeM6DVvnKctCLSXb1n2gYlslGsMEGcoCuaDVhytaapida0pWFIMxVRDkPv6o7GWc2ETQGb67Q+h0BS+5SBuE87Zzi-69AGC4tkZG70MRxYsDOw8OGPQ7jr2GZSDmhb-V4RY2tKXG5kG-07jVCBTKWGBWooUnsdRCmqcW9DLtnC5yicEJ8RJ4a0FGkJxr+8mVJgOMceUkWcQgcXxFCKW0ha0aOBBwYldm+ahfVklrk17Xgkl4n9a5uWFcm-mQDN6m5BGZlZYoNWdqt0WtYlvX-SPC6XbRqUPYO226b9hnLZ9GNvKwEOHc57nUl5k23ZjtAvdIH36avM5A4ZNOdftkmiZJo2ybzynzZp8R49Sf30xFiv05r8OxomqOBfNjvoXZoMQSMFHZSVzG+dLz5x60SemnEXUlYDZCR6lGF8py+51Odgetru92m4Wrflu+gZ1pgTaAe+8cHqezmnpPSGgZYeHtp+2gRSjwHv4NUqE0FIcwd6higJaKG9oMRf3TDvN8Wx6A6maH3bA4tYCmzPhbF08onryDQF5Hy0YZaH0VvfD6Q9z6J1WitE+N8756AAYAmYxtZwoHwXIQh9JvJRgvifboj1nqvzelDW6HxfquyYQ-aObCEE1H4S6dBiZpK0zSHLK279v7Q1WrDT6883R4jgjOEIoZgi1zSPXPm6NlaqxoemQxHptyhDMWHDO8sj6N0FlvLEMEjHTmJMnIybMOYGx5lY3ONj86F2LgnAxfinFekDsEkyRILE5ywYLFutA27-XibFRJJjkms1STOdJHjyE2NSHY-JsEAkizjuonmcg6jQQKcYhS+han+OcSLAy+Y6IlLHrVKhOCHEJOMc5UJTtwnG0yc3GJGBfbt3sR8RxkzSluLrnMrxzdGnLLyehdZATNnlMjkrapc8jkTICaQnmFSpEsGYf1bB-C6EugYahChgDZHHTaXU7cij0yCOfqkERWjtpwNWkKX+lSnkyNYX8+RXzxHpmUbAbJuTAlXAhatHRko9G2nnjeE5LlsJhOzhE+ZntvZLJLuhElzjwFbMsTsqJ2DMVNJWcSycpLYSGT7g8wetirnKUZUSFpk87lgr2AABh5ZuAJBMoLUmnm7WeucFUAUIXiE0WhZQHQpWkdFwrom0oOT4jMvLnGcFFDUfVmp0kmouSrUG+z6ViutUSJBKD0lUt2dQ+e2BYA1GMXsAATPPFetBdRoAAF6MhWMRHU7UBiSr3siA5nT56YQwNhI0sx8JpFOX3f17LvGrKlLm-NuEi1TLSYK85btLmavQtWnChaCLKuLajGeAVRUunbQWvCBFbV6uaI6tBMoVEupqW2k8eaO0jrSPicQaQzxlT9Wy6OrzK1poXTWztK6zHrt4eeE6jbPHluHnujCB60AAEdXiDBGhgZ+zKjUZIDQdRZFrb3tqfYyV976yUsq-de6m7q4nzqwo+59wGwGgbOVeqULbLWqXZlqZBGoLGppUisZstscphhjJ0DSWlHZZyjn1cSKA5Dy2+E0LgTgE30mpWMj4WA6MMZ+P0lIwRWNazw0QqMU9hUastczaVEcUPPPTGaoudLoMfXeemT5x9v73SES-V6uLJRQpPpI-+CL3ZyJchDd62iYGOnnqVcesxGohAIBdKWfZ8CxynRg4VaHb12e+A5hMTmeIbvHluhuVTXWtuUn53iMhAvOZToyc9YGhWzoHcFXKoVx4whCNgZLYXrE7orbZzLGUwa5YJpuy9cLRmWpi0mGiUBQyIfo3lsqzUG0OzwzF8gIH+WZ0Ntu2rvnStlV6y1lpKWm0RbndF0b4903SYUvK9CMWIowCijFRyB1VXif7VFtK83MobeyrFeKY77UTo8xnZ1zbIt1aO4QrKtw8pvl232rGB2MuyTKutzbOVXsIk2csWo04lbRRQPZB7P3x6PEqu99V+3LWdTAb2xHn3LVmleGgMCg5BjAgRzYiTt6sdNL65Nz9ZaisLPNR6l0pPfbk4FRnKnozOV-vngz8bYNJvIZqyKr7HwudvrmIt-er6ZWollScB0hQQCFCAA).

You can space your operators to work around this issue, e.g. `height: 1px * @lineHeight;`.

The underlying bug can either be solved by patching [the upstream parser](https://github.com/shellscape/postcss-less) or by the addition of a `isStandardSyntaxUnit` utility function here in stylelint that ignores "units" that contain operators, e.g. `*`, `/`, `+` and `-`.

I've labelled the issue as ready to implement. Please consider [contributing](https://stylelint.io/contributing) if you have time.

There are [steps on how to fix a bug in a rule](https://stylelint.io/developer-guide/rules#fix-a-bug-in-a-rule) in the Developer guide.

**Comment by superchow** [comment]
> @superchow Thanks for the report and for using the template.
> 
> It looks like it's the `unit-case` rule causing the issue. See [this demo](https://stylelint.io/demo#N4Igxg9gJgpiBcIB0BXAtASwHYAcUBcAKAAQBtsYAJGDAcwAt8AaAAmJwEMoptaAVCDlbsuPLLQBCEfPggBbAJQtgAHSwsW9Gg3zwWARhwAPAFRkK1OowDcajZ2680snHsOmRj8QJy31LBzFaNAAjaVk5N2MzQN4pGXk-DXIsGDQtK10DaPNUyx0-AF81ECYQAGcATyx8DiMEEFIYcvLS8AgsADM6BtV-FRAAJxQm8oG9Po0NAY58NGGmtCwINBQsAGtlgHcscZZ8YZgmOymBkNIIMHWllZg5HHxKvYOUI5Pp9ovBm8wsADcOOQoOkYEZnodjv4PpA5HIYDUfncHk8EPsIe8WANYGBSBxBrMMB1Qhcrj8oCgcOQwLM0jhBoIYIN8BhmnsANoYjQvN5QqaTKYCgZ0ZaDGDszkCzHtLDlGBgAgYP5pcmUjDU-DNNBbDD4ehoHidTqM+FzAGkV5jEASjQAXQlxV5NshgpA2Nx+OZRPOl2uyzQ5XoECZ9A4WGBdIZTMqaAgSsGgwwsEtem5ztOIE6HTmnQ4cgwpGjfpVVJpS1zrNRqYxA0zCJzeYLPzzLSctHhjLVaHr+ej6xglS2Qag4NeaY+nTWYE9WDQ1NIYB+2DNiZHPJdE6wU8JM7nC79a3KnDAMGBkdmQdXY6lG63RJSMDxaFo+J4Jp+yxltTDeOBPFFt92St0V5AY+0qTp8ThfU5XdAkiT9DB7iDL98EvasQDhHgOC7B98BQUUyyg-cNm2QCU2Al0i2aY8wycQ85QwbowB1FFyNHdCiwpEsNTQWZMCQpl5hGCs2LXdNONVdU0llJopyDZM0XYkCQD9JFHn9CB8OPNDlNUowDmw2U80gC4ZR0yiViXQFE31TTzmk3EA1neQ4RqBSq2UiMcEZdTiM2CAdnM9MZLlWRvhwWUUCgFYcQ4FofjWfzAqApSXRCuTwsi6K0BgJpXLmPzSKCj50rC5xKm8hKSICsjFLEkqDicP1Ui2e9iqlNYdSqpLao8l0+IWNI1Oje9QhgTNRXFXkPkBLYOEqS0rw0flJSlUFjweKbVpdb0rlGcpeM6DVvnKctCLSXb1n2gYlslGsMEGcoCuaDVhytaapida0pWFIMxVRDkPv6o7GWc2ETQGb67Q+h0BS+5SBuE87Zzi-69AGC4tkZG70MRxYsDOw8OGPQ7jr2GZSDmhb-V4RY2tKXG5kG-07jVCBTKWGBWooUnsdRCmqcW9DLtnC5yicEJ8RJ4a0FGkJxr+8mVJgOMceUkWcQgcXxFCKW0ha0aOBBwYldm+ahfVklrk17Xgkl4n9a5uWFcm-mQDN6m5BGZlZYoNWdqt0WtYlvX-SPC6XbRqUPYO226b9hnLZ9GNvKwEOHc57nUl5k23ZjtAvdIH36avM5A4ZNOdftkmiZJo2ybzynzZp8R49Sf30xFiv05r8OxomqOBfNjvoXZoMQSMFHZSVzG+dLz5x60SemnEXUlYDZCR6lGF8py+51Odgetru92m4Wrflu+gZ1pgTaAe+8cHqezmnpPSGgZYeHtp+2gRSjwHv4NUqE0FIcwd6higJaKG9oMRf3TDvN8Wx6A6maH3bA4tYCmzPhbF08onryDQF5Hy0YZaH0VvfD6Q9z6J1WitE+N8756AAYAmYxtZwoHwXIQh9JvJRgvifboj1nqvzelDW6HxfquyYQ-aObCEE1H4S6dBiZpK0zSHLK279v7Q1WrDT6883R4jgjOEIoZgi1zSPXPm6NlaqxoemQxHptyhDMWHDO8sj6N0FlvLEMEjHTmJMnIybMOYGx5lY3ONj86F2LgnAxfinFekDsEkyRILE5ywYLFutA27-XibFRJJjkms1STOdJHjyE2NSHY-JsEAkizjuonmcg6jQQKcYhS+han+OcSLAy+Y6IlLHrVKhOCHEJOMc5UJTtwnG0yc3GJGBfbt3sR8RxkzSluLrnMrxzdGnLLyehdZATNnlMjkrapc8jkTICaQnmFSpEsGYf1bB-C6EugYahChgDZHHTaXU7cij0yCOfqkERWjtpwNWkKX+lSnkyNYX8+RXzxHpmUbAbJuTAlXAhatHRko9G2nnjeE5LlsJhOzhE+ZntvZLJLuhElzjwFbMsTsqJ2DMVNJWcSycpLYSGT7g8wetirnKUZUSFpk87lgr2AABh5ZuAJBMoLUmnm7WeucFUAUIXiE0WhZQHQpWkdFwrom0oOT4jMvLnGcFFDUfVmp0kmouSrUG+z6ViutUSJBKD0lUt2dQ+e2BYA1GMXsAATPPFetBdRoAAF6MhWMRHU7UBiSr3siA5nT56YQwNhI0sx8JpFOX3f17LvGrKlLm-NuEi1TLSYK85btLmavQtWnChaCLKuLajGeAVRUunbQWvCBFbV6uaI6tBMoVEupqW2k8eaO0jrSPicQaQzxlT9Wy6OrzK1poXTWztK6zHrt4eeE6jbPHluHnujCB60AAEdXiDBGhgZ+zKjUZIDQdRZFrb3tqfYyV976yUsq-de6m7q4nzqwo+59wGwGgbOVeqULbLWqXZlqZBGoLGppUisZstscphhjJ0DSWlHZZyjn1cSKA5Dy2+E0LgTgE30mpWMj4WA6MMZ+P0lIwRWNazw0QqMU9hUastczaVEcUPPPTGaoudLoMfXeemT5x9v73SES-V6uLJRQpPpI-+CL3ZyJchDd62iYGOnnqVcesxGohAIBdKWfZ8CxynRg4VaHb12e+A5hMTmeIbvHluhuVTXWtuUn53iMhAvOZToyc9YGhWzoHcFXKoVx4whCNgZLYXrE7orbZzLGUwa5YJpuy9cLRmWpi0mGiUBQyIfo3lsqzUG0OzwzF8gIH+WZ0Ntu2rvnStlV6y1lpKWm0RbndF0b4903SYUvK9CMWIowCijFRyB1VXif7VFtK83MobeyrFeKY77UTo8xnZ1zbIt1aO4QrKtw8pvl232rGB2MuyTKutzbOVXsIk2csWo04lbRRQPZB7P3x6PEqu99V+3LWdTAb2xHn3LVmleGgMCg5BjAgRzYiTt6sdNL65Nz9ZaisLPNR6l0pPfbk4FRnKnozOV-vngz8bYNJvIZqyKr7HwudvrmIt-er6ZWollScB0hQQCFCAA).
> 
> You can space your operators to work around this issue, e.g. `height: 1px * @lineHeight;`.
> 
> The underlying bug can either be solved by patching [the upstream parser](https://github.com/shellscape/postcss-less) or by the addition of a `isStandardSyntaxUnit` utility function here in stylelint that ignores "units" that contain operators, e.g. `*`, `/`, `+` and `-`.
> 
> I've labelled the issue as ready to implement. Please consider [contributing](https://stylelint.io/contributing) if you have time.
> 
> There are [steps on how to fix a bug in a rule](https://stylelint.io/developer-guide/rules#fix-a-bug-in-a-rule) in the Developer guide.

`height: 1px * @lineHeight;` check pass. But it not work after `--fix`.

**Comment by gmenih341** [comment]
I'm having a similar issue, I have a `@function assetUrl`, which when used, always gets lowercased when using `--fix`.

this:

```
background-image: assetUrl('assets/images/some-image.jpg');
```

is always fixed to:
```
background-image: asseturl('assets/images/campanda-renter-image.jpg');
```

**Comment by jeddy3** [comment]
@gmenih341 That's the intended behaviour. You can use the [`ignoreFunctions: []` secondary option](https://stylelint.io/user-guide/rules/function-name-case#ignorefunctions-regex-as-string-regex-non-regex) of the `function-name-case` rule if you want `assetUrl` to be a special case.

**Comment by jeddy3** [comment]
Closing [as stylistic rules are frozen](https://github.com/stylelint/stylelint/issues/5674). The community is welcome to migrate the rule [to a plugin](https://stylelint.io/developer-guide/plugins) and fix this bug.
